deepchem/models/tensorgraph/tensor_graph.py

```diff
@@ -501,14 +532,18 @@ def build(self):
       writer.close()
 
     # As a sanity check, make sure all tensors have the correct shape.
+    # Don't check this for now
 
     for layer in self.layers.values():
-      try:
-        assert list(layer.shape) == layer.out_tensor.get_shape().as_list(
-        ), '%s: Expected shape %s does not match actual shape %s' % (
-            layer.name, layer.shape, layer.out_tensor.get_shape().as_list())
-      except NotImplementedError:
-        pass
+      # Dequeued objects from a queue may not necessarily have a well-defined
+      # shape.
+      if layer.shape is not None:
+        try:
+          assert list(layer.shape) == layer.out_tensor.get_shape().as_list(
+          ), '%s: Expected shape %s does not match actual shape %s' % (
+              layer.name, layer.shape, layer.out_tensor.get_shape().as_list())
+        except NotImplementedError:
+          pass
```

Comment:
This change too is unnecessary.  It already handles the case where the shape is unknown.  That's signaled by an exception rather than returning `None`.

Reply:
Okay there is so much inconsistency between when _shape and shape is used. Soemtimes shape the property is invoked, other times _shape is invoked. There's so much cruft that refers to _shape that I'd need to guard literally every shape call with try: except NotImplemented.

Simply put, this code was not designed at all with _shape being set to None in mind.

Reply:
Where is there code that refers to `_shape` directly?  It's a private field (hence it starts with `_`), and it didn't exist at all until a couple of months ago.  The only place `_shape` should ever be referenced directly is within a Layer class.  All outside code should use the public accessor `shape`.